data/src/main/java/com/teamagam/gimelgimel/data/rasters/repository/IntermediateRasterVisibilityDataRepository.java

```diff
@@ -0,0 +1,60 @@
+package com.teamagam.gimelgimel.data.rasters.repository;
+
+import com.teamagam.gimelgimel.data.base.repository.ReplayRepository;
+import com.teamagam.gimelgimel.domain.rasters.entity.IntermediateRasterVisibilityChange;
+import com.teamagam.gimelgimel.domain.rasters.repository.IntermediateRasterVisibilityRepository;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import rx.Observable;
+
+public class IntermediateRasterVisibilityDataRepository implements IntermediateRasterVisibilityRepository {
+
+    private final ReplayRepository<IntermediateRasterVisibilityChange> mInnerRepo;
+    private String mCurrentlySelectedName;
+
+    @Inject
+    public IntermediateRasterVisibilityDataRepository() {
+        mInnerRepo = ReplayRepository.createReplayAll();
```

Comment:
Why not replay(1)? Is that necessary?
If we have many events, the ViewModel will select items in order (Replay all the actions again).

SubjectFactory should be used here.